pallets/oracle/src/dia.rs

```diff
@@ -0,0 +1,207 @@
+	}
+}
+
+pub struct MockDiaOracleConvertor;
+
+impl Convert<OracleKey, Option<(Vec<u8>, Vec<u8>)>> for MockDiaOracleConvertor {
+	fn convert(spacwalk_oracle_key: OracleKey) -> Option<(Vec<u8>, Vec<u8>)> {
+		match spacwalk_oracle_key {
+			OracleKey::ExchangeRate(currency_id) => match currency_id {
+				CurrencyId::Token(token_symbol) => match token_symbol {
+					primitives::TokenSymbol::DOT => return Some((vec![0u8], vec![1u8])),
+					primitives::TokenSymbol::PEN => return Some((vec![0u8], vec![2u8])),
+					primitives::TokenSymbol::KSM => return Some((vec![0u8], vec![3u8])),
+					primitives::TokenSymbol::AMPE => return Some((vec![0u8], vec![4u8])),
+				},
+				CurrencyId::ForeignAsset(foreign_asset_id) =>
+					Some((vec![1u8], foreign_asset_id.to_le_bytes().to_vec())),
+				CurrencyId::Native => Some((vec![2u8], vec![])),
+				CurrencyId::StellarNative => Some((vec![3u8], vec![])),
+				CurrencyId::AlphaNum4 { code, .. } => Some((vec![4u8], code.to_vec())),
+				CurrencyId::AlphaNum12 { code, .. } => Some((vec![5u8], code.to_vec())),
+			},
+			OracleKey::FeeEstimation => Some((vec![6u8], vec![])),
+		}
+	}
+}
+
+impl Convert<(Vec<u8>, Vec<u8>), Option<OracleKey>> for MockDiaOracleConvertor {
+	fn convert(dia_oracle_key: (Vec<u8>, Vec<u8>)) -> Option<OracleKey> {
+		let (blockchain, symbol) = dia_oracle_key;
+		match blockchain[0] {
+			0u8 => match symbol[0] {
+				1 =>
+					return Some(OracleKey::ExchangeRate(CurrencyId::Token(
+						primitives::TokenSymbol::DOT,
+					))),
+				2 =>
+					return Some(OracleKey::ExchangeRate(CurrencyId::Token(
+						primitives::TokenSymbol::PEN,
+					))),
+				3 =>
+					return Some(OracleKey::ExchangeRate(CurrencyId::Token(
+						primitives::TokenSymbol::KSM,
+					))),
+				4 =>
+					return Some(OracleKey::ExchangeRate(CurrencyId::Token(
+						primitives::TokenSymbol::AMPE,
+					))),
+				_ => return None,
+			},
+			1u8 => {
+				let x = [symbol[0], symbol[1], symbol[2], symbol[3]];
+				let number = u32::from_le_bytes(x);
+				Some(OracleKey::ExchangeRate(CurrencyId::ForeignAsset(number)))
+			},
+			2u8 => Some(OracleKey::ExchangeRate(CurrencyId::Native)),
+			3u8 => Some(OracleKey::ExchangeRate(CurrencyId::StellarNative)),
+			4u8 => {
+				let vector = symbol;
+				let code = [vector[0], vector[1], vector[2], vector[3]];
+				Some(OracleKey::ExchangeRate(CurrencyId::AlphaNum4 { code, issuer: [0u8; 32] }))
+			},
+			5u8 => {
+				let vector = symbol;
+				let code = [
+					vector[0], vector[1], vector[2], vector[3], vector[4], vector[5], vector[6],
+					vector[7], vector[8], vector[9], vector[10], vector[11],
+				];
+				Some(OracleKey::ExchangeRate(CurrencyId::AlphaNum12 { code, issuer: [0u8; 32] }))
+			},
+			6u8 => Some(OracleKey::FeeEstimation),
+			_ => None,
+		}
+	}
+}
+pub struct MockConvertPrice;
+impl Convert<u128, Option<FixedU128>> for MockConvertPrice {
+	fn convert(price: u128) -> Option<FixedU128> {
+		Some(FixedU128::from_inner(price))
+	}
+}
+
+pub struct MockMoment;
```

Comment:
I'd prefer a different name for this. Maybe similar to the others `MockConvertMoment`

Reply:
Done